tests/testthat/test-get_vpts.R

```diff
@@ -0,0 +1,425 @@
+      n_dbz_all = "integer",
+      rcs = "numeric",
+      sd_vvp_threshold = "numeric",
+      vcp = "integer",
+      radar_latitude = "numeric",
+      radar_longitude = "numeric",
+      radar_height = "integer",
+      radar_wavelength = "numeric",
+      source_file = "character"
+    )
+
+  expect_identical(
+    get_vpts(radar = c("deflg"),
+             date = lubridate::ymd("20171015"),
+             source = "baltrad",
+             as_tibble = TRUE) |>
+      purrr::map(class),
+    expected_col_types
+  )
+
+  # Specific radar causing trouble
+  expect_identical(
+    get_vpts(radar = c("dehnr"),
+             date = lubridate::ymd("20171015"),
+             source = "uva",
+             as_tibble = TRUE) |>
+      purrr::map(class),
+    expected_col_types
+  )
+})
+
+test_that("get_vpts() can fetch data from a specific source only", {
+  skip_if_offline()
+
+  # Data from only BALTRAD even if UVA is available for the same interval
+  expect_identical(
+    get_vpts(radar = "bejab",
+             date = "2018-05-18",
+             source = "baltrad",
+             as_tibble = TRUE) |>
+      dplyr::pull("source") |>
+      unique(),
+    "baltrad"
+  )
+})
+
+test_that("get_vpts() can fetch vpts data for a date range", {
+  skip_if_offline()
+
+  radar_interval <- get_vpts(radar = "bejab",
+                             lubridate::interval(
+                               lubridate::ymd("2023-01-01"),
+                               lubridate::ymd("2023-01-02")
+                             ),
+                             source = "baltrad",
+                             as_tibble = TRUE
+  )
+  expect_s3_class(
+    radar_interval,
+    "data.frame"
+  )
+
+  # Check that the requested dates are present in the output
+  expect_in(
+    unique(as.Date((radar_interval$datetime))),
+    c(as.Date("2023-01-01"), as.Date("2023-01-02"))
+  )
+
+})
+
+test_that("get_vpts() supports date intervals with hours and minutes",{
+  skip_if_offline()
+
+  hour_minute_interval <-
+    get_vpts(radar = "bejab",
+             lubridate::interval(
+               lubridate::ymd_hms("2023-01-01 12:00:00"),
+               lubridate::ymd_hms("2023-01-01 16:59:59")
+             ),
+             source = "baltrad",
+             as_tibble = TRUE
+    )
+
+  expect_s3_class(
+    hour_minute_interval,
+    "data.frame"
+  )
+
+  # The minimum returned date should be within the interval
+  expect_gte(
+    min(hour_minute_interval$datetime),
+    lubridate::ymd_hms("2023-01-01 12:00:00")
+  )
+  # The maximum returned datetime should be within the interval
+  expect_lte(
+    max(hour_minute_interval$datetime),
+    lubridate::ymd_hms("2023-01-01 16:59:59")
+  )
+
+  # The maximum should actually be rounded by a 15 minute interval
+  expect_identical(
+    max(hour_minute_interval$datetime),
+    lubridate::ymd_hms("2023-01-01 16:45:00")
+  )
+})
+
+test_that("get_vpts() can return data as a vpts object compatible with getRad",{
```

Comment:
This test takes a long time, and is possibly causing the windows machine to fail on CI due to a timeout

Reply:
Fixed in https://github.com/aloftdata/getRad/blob/228d26d145bf65c50ceff3a53a89fe25d6986c2c/tests/testthat/test-get_vpts.R#L261